### PR TITLE
Migrate UniqueArrayListTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate UniqueArrayListTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt [hunk 1]
```diff
@@ -13,574 +13,466 @@
...
+        "55",
+        "TEst",
+        "12",
+        "dsf23A",
+        "23",
+        "sd"
+    )
+
+    private inline fun <reified E> assertNotSameLists(a: MutableList<E>, b: MutableList<E>) {
+        MatcherAssert.assertThat(b, not(IsIterableContainingInOrder.contains<Any>(*a.toTypedArray())))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt:60]: import `MatcherAssert`

File: AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt [hunk 2]
```diff
@@ -13,574 +13,466 @@
...
-
-        uniqueList.clear();
-        Collections.reverse(list);
-        uniqueList.addAll(list);
-
-        assertEquals(1, uniqueList.size());
-        assertEquals("tarekkma", uniqueList.get(0));
+    fun test_comparator() {
+        val list = Arrays.asList("TarekkMA", "TarekkMA", "TarekkmA", "tarekkma")
+        val uniqueList = from(list, java.lang.String.CASE_INSENSITIVE_ORDER)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt:111]: Unsure if you can change this to `String.CASE_INSENSITIVE_ORDER` as it is in the Java - probably not

File: AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt [hunk 3]
```diff
@@ -13,574 +13,466 @@
...
-        try (MockedStatic<Collections> MockCollection = mockStatic(Collections.class)) {
-            UniqueArrayList<Integer> uniqueList = UniqueArrayList.from(longs);
-            uniqueList.sort();
-            assertListEquals(sorted, uniqueList);
-
-            MockCollection.verify(() -> Collections.sort(any(), any()), never());
+package com.ichi2.utils
+
+import com.ichi2.utils.ListUtil.Companion.assertListEquals
+import com.ichi2.utils.UniqueArrayList.Companion.from
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt:19]: remove this import. Change callers to `UniqueArrayList.from`

**Review by david-allison — CHANGES_REQUESTED** [review]
Could you add the newlines back into the class - lots of changes there which reduce the readability of the tests

File: AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+        assertListEquals(longs, uniqueList)
+
+        uniqueList.add(575757)
+        assertNotSameLists(longs, uniqueList)
+    }
+
+    @Test
+    fun test_comparator() {
+        val list = Arrays.asList("TarekkMA", "TarekkMA", "TarekkmA", "tarekkma")
+        val uniqueList = UniqueArrayList.from(list, String.CASE_INSENSITIVE_ORDER)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt:118]: awesome!

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt (hunk 4 above)
Comment (oyeraghib) [reply AnkiDroid/src/test/java/com/ichi2/utils/UniqueArrayListTest.kt:118]: thanks!! somehow it worked referencing to string.kt

**Review by mikehardy — APPROVED** [review]
:100:

**Pull request merged**